scripts/microbe_stage/microbe_ai.as

```diff
@@ -1,927 +1,1133 @@
+
+////////////////////////////////////////////////////////////////////////////////
+// MicrobeAIControllerComponent
+//
+// Component for identifying and determining AI controlled microbes.
+////////////////////////////////////////////////////////////////////////////////
+
+// Enum for state machine
+    enum LIFESTATE
+        {
+        NEUTRAL_STATE,
+        GATHERING_STATE,
+        FLEEING_STATE,
+        PREDATING_STATE,
+        PLANTLIKE_STATE,
+        SCAVENGING_STATE
+        }
+
+class MicrobeAIControllerComponent : ScriptComponent{
+
+    MicrobeAIControllerComponent(){
+
+        intervalRemaining = reevalutationInterval;
+    }
+
+    float movementRadius = 2000;
+    // That means they evaluate every AI_CELL_THINK_INTERVAL seconds
+    float reevalutationInterval = AI_CELL_THINK_INTERVAL;
+    float intervalRemaining;
+    int boredom = 0;
+    //! This needs to be changed to use elapsed time instead of elapsed frames
+    int ticksSinceLastToggle = 600;
+    double previousStoredCompounds = 0.0f;
+    Float3 direction = Float3(0, 0, 0);
+    double speciesAggression = -1.0f;
+    double speciesFear = -1.0f;
+    double speciesActivity = -1.0f;
+    double speciesFocus = -1.0f;
+    double speciesOpportunism = -1.0;
+    bool hasTargetPosition = false;
+    Float3 targetPosition = Float3(0, 0, 0);
+    bool hasSearchedCompoundId = false;
+    CompoundId searchedCompoundId;
+    float previousAngle = 0.0f;
+    float compoundDifference=0;
+    ObjectID prey = NULL_OBJECT;
+    ObjectID targetChunk = NULL_OBJECT;
+    ObjectID predator = NULL_OBJECT;
+    bool moveThisHunt = true;
+    bool moveFocused = false;
+    bool preyPegged=false;
+    // Prey and predator lists
+    array<ObjectID> predatoryMicrobes;
+    array<ObjectID> preyMicrobes;
+    array<ObjectID> chunkList;
+	bool aiInitialized = false;
+    LIFESTATE lifeState = NEUTRAL_STATE;
+
+}
+
+
+//! \todo Check if there is a better way than caching a single component for this system
+class MicrobeAISystemCached{
+
+    MicrobeAISystemCached(ObjectID entity, MicrobeAIControllerComponent@ first,
+        MicrobeComponent@ second, Position@ third
+    ) {
+
+        this.entity = entity;
+        @this.first = first;
+        @this.second = second;
+        @this.third = third;
+    }
+
+    ObjectID entity = -1;
+
+    MicrobeAIControllerComponent@ first;
+    MicrobeComponent@ second;
+    Position@ third;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// MicrobeAISystem
+//
+// Updates AI controlled microbes
+////////////////////////////////////////////////////////////////////////////////
+class MicrobeAISystem : ScriptSystem{
+
+    void Init(GameWorld@ w){
+
+        @this.world = cast<CellStageWorld>(w);
+        passedTime = 0;
+        assert(this.world !is null, "MicrobeAISystem expected CellStageWorld");
+    }
+
+    void Release(){}
+
+    void Run(float elapsed){
+        passedTime += elapsed;
+		ForceRun(elapsed);

```

Comment:
```suggestion
        if(!aiComponent.aiInitialized)
        {
            aiComponent.intervalRemaining = aiComponent.reevaluationInterval;
            aiComponent.aiInitialized = true;
        }
```
I think this would be much more effective (as the current method prevents the AI from doing anything at all).